test/python/http_recorder.py

```diff
@@ -0,0 +1,285 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""Utilities (based on VCRpy) to record remote requets and allow testing offline/cached."""
+
+import json
+from vcr.persisters.filesystem import FilesystemPersister
+from vcr import VCR
+
+
+class IdRemoverPersister(FilesystemPersister):
+    """
+    IdRemoverPersister is a VCR persister. This is, it implements a way to save and load cassettes.
+    This persister in particular inherits load_cassette from FilesystemPersister (basically, it
+    loads a standard cassette in the standard way from the FS). On the saving side, it removes some
+    fields in the JSON content of the responses for dummy values.
+    """
+
+    @staticmethod
+    def get_responses_with(string2find, cassette_dict):
+        """
+        Filters the requests from cassette_dict
+
+        Args:
+            string2find (str): request path
+            cassette_dict (dict): a VCR cassette dictionary
+
+        Returns:
+            Request: VCR's representation of a request.
+        """
+        requests_indeces = [i for i, x in enumerate(cassette_dict['requests']) if
+                            string2find in x.path]
+        return [cassette_dict['responses'][i] for i in requests_indeces]
+
+    @staticmethod
+    def get_new_id(field, path, id_tracker, _type=str):
```

Comment:
According to PEP8, it is preferable if you use `type_`.

Reply:
d839c20